src/main/webapp/app/course/learning-paths/pages/learning-path-student-page/learning-path-student-page.component.ts

```
@@ -32,45 +32,49 @@
+    async startLearningPath(): Promise<void> {
         try {
-            this.isLearningPathIdLoading.set(true);
-            const learningPathId = await this.learningApiService.generateLearningPath(courseId);
-            this.learningPathId.set(learningPathId);
+            this.isLearningPathLoading.set(true);
+            if (!this.learningPath()) {
+                const learningPath = await this.learningApiService.generateLearningPathForCurrentUser(this.courseId());
+                this.learningPath.set(learningPath);
+            }
+            await this.learningApiService.startLearningPathForCurrentUser(this.learningPath()!.id);
+            this.learningPath.update((learningPath) => ({ ...learningPath!, startedByStudent: true }));
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)